### PR TITLE
Update .Net Core runtime from 3.0-preview6 to preview7

### LINKED ISSUE #9998: Active Directory RSAT Get-ADUser issue
# Steps to reproduce
Install Optional Features to enable RSAT for Active Directory, reboot, then run PSCore6 and test AD cmdlets by doing "Get-ADUser _username_"  - this should show you your NT acct details. If that does work without issue (indicating the modules loaded as expected) you can reproduce my error by running

Get-ADUser _username_ -Properties *

# Expected behavior

You should get an export of all your AD acct properties like below (taken from Windows Powershell - which works as expected)

image

# Actual behavior

image

# Environment data

PS C:\Users\nmyron> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.2.1
PSEdition                      Core
GitCommitId                    6.2.1
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

### PR DESCRIPTION
**Pull request by bergmeister**
# PR Summary

Update also the out of band NuGet packages to preview7 as well except the ones that could not be updated in the past (System.ServiceProcess.ServiceController and the WindowsCompatibility package that has a dependency on it, issue #10067 is tracking that)
Notes: .Net Core Runtime does not ship SOS dll any more by default (out of band now) and it seems they've removed a few localised DLLs as well (probably as part of their SDK size optimisation). The SOS dll is now an optional install addon to the dotnet cli and not a NuGet package. I'm not 100% sure if we can just stop shipping that DLL or some people have a dependency on it somehow?

Preview7 now requires `<UseWpf>` and `<UseWindowsForms>` to be set to `true` in the csproj, so added them conditionally if using `WindowsDesktop` SDK.

Fixes #9998 (manually tested)

# PR Context

https://devblogs.microsoft.com/dotnet/announcing-net-core-3-0-preview-7
https://github.com/dotnet/core/blob/master/release-notes/3.0/preview/3.0.0-preview7.md

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - SOS DLL not shipped any more. Should we re-add it for back-compat?
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
```
Comment (PoshChan) [inline .devcontainer/Dockerfile]: @bergmeister, your last commit had 1 failures in `PowerShell-CI-windows`
Validate Update-Help from the Web for one PowerShell module..Validate Update-Help for module 'Microsoft.PowerShell.Core' with scope as 'False'
```powershell
Expected 1, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Help\UpdatableHelpSystem.Tests.ps1: line 181
181:     $helpFilesInstalled.Count | Should -Be $expectedHelpFiles.Count
```

Comment (bergmeister) [reply .devcontainer/Dockerfile]: It seems to be a sporadic failure, will issue an empty commit to retry

**Comment by iSazonov** [comment]
We could set `<LangVersion>preview</LangVersion>`.

**Review by iSazonov — APPROVED** [review]

**Comment by bergmeister** [comment]
@iSazonov According to the docs, this should already be the default but even if not, I'd prefer a separate PR for that as it seems unrelated
https://docs.microsoft.com/dotnet/csharp/language-reference/configure-language-version

**Comment by daxian-dbw** [comment]
The SOS dll is used when it's time to debug powershell using WinDbg. It's useful to keep it there.

> The SOS dll is now an optional install addon to the dotnet cli and not a NuGet package.

Can you please point me to the instructions about installing SOS dll?

**Comment by bergmeister** [comment]
@daxian-dbw Installation instructions for SOS are here now:
https://github.com/dotnet/diagnostics#installing-sos

**Comment by bergmeister** [comment]
NB: I now also update the out of band NuGet packages to preview7 as well since the tests are green with it. This applies to all out of band Nuget packages except the ones that could not be updated in the past for preview6 (System.ServiceProcess.ServiceController and the WindowsCompatibility package that has a dependency on it, issue #10067 is tracking that)

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (PoshChan) [inline .devcontainer/Dockerfile]: @bergmeister, your last commit had 1 failures in `PowerShell-CI-windows`
ConsoleHost unit tests.ApartmentState WPF tests.WPF requires STA and will work
```powershell
The following exception occurred while retrieving member "Load": "Could not load type 'MS.Internal.SecurityCriticalDataForSet`1' from assembly 'WindowsBase, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'."
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\ConsoleHost.Tests.ps1: line 733
733:             $Window = [System.Windows.Markup.XamlReader]::Load($reader)

```

Comment (bergmeister) [reply .devcontainer/Dockerfile]: Probably sporadic failure, will make empty commit to retry (did not see this failure in my own Azure Pipelines builds)

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @bergmeister, your last commit had 1 failures in `PowerShell-CI-windows`
ConsoleHost unit tests.ApartmentState WPF tests.WPF requires STA and will work
```powershell
The following exception occurred while retrieving member "Load": "Could not load type 'MS.Internal.SecurityCriticalDataForSet`1' from assembly 'WindowsBase, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'."
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\ConsoleHost.Tests.ps1: line 733
733:             $Window = [System.Windows.Markup.XamlReader]::Load($reader)

```

**Comment by daxian-dbw** [comment]
@bergmeister Does the service cmdlet failure still persist even with the `preview7` version of the `System.ServiceProcess.ServiceController` NuGet package?

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (bergmeister) [reply .devcontainer/Dockerfile]: I can reproduce this locally, the constructor of `System.Windows.Markup.XamlReader` throws, which is probably a new WPF bug, will dig further
```pwsh
add-type -AssemblyName presentationframework
$xaml = [xml]@"
            <Window
                xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
                x:Name="Window" Title="Initial Window" WindowStartupLocation = "CenterScreen"
                Width = "400" Height = "300" ShowInTaskbar = "True">
            </Window>
"@
$reader = [System.Xml.XmlNodeReader]::new($xaml)
[System.Windows.Markup.XamlReader]::Load($reader)
```

**Comment by bergmeister** [comment]
@daxian-dbw Yes, see the test build [here](https://christophbergmeister.visualstudio.com/PowerShell/_build/results?buildId=1231&view=ms.vss-test-web.build-test-results-tab). I have even tried older preview version (back to preview3) of .Net Core 3 and all those packages cause the same 2 failures. @iSazonov suspects it might be the GC: https://github.com/PowerShell/PowerShell/issues/10067#issuecomment-508948459

After the STA commit got merged into master recently, the newly added tests is failing now for preview7, I provided lots of details in my comment [here](https://github.com/PowerShell/PowerShell/pull/10227#discussion_r307960566) and it is again not easy to isolate the problem into a simple repro that we could give the WPF team but maybe just the error message could help them find out what goes wrong

**Comment by SteveL-MSFT** [comment]
The static analysis failure is fixed via https://github.com/PowerShell/PowerShell/pull/10297

Doing a manual build of your branch, there is a warning that `<UseWpf>` and `<UseWindowsForms>` needs to be explicitly set to true.  This appears to be a change from Preview6 where it was implicit when using the DesktopSDK.  Adding it conditionally fixes the type exception.

**Review by SteveL-MSFT — APPROVED** [review]

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 3 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.http://powershell.com/cs/blogs/ebookv2/default.aspx should work
```powershell
retry of URL failed with error: 
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 111
111:                                 throw "retry of URL failed with error: $($_.Message)"

```
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/README.md.http://powershell.com/cs/blogs/ebookv2/default.aspx should work
```powershell
retry of URL failed with error: 
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 111
111:                                 throw "retry of URL failed with error: $($_.Message)"

```
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/README.md.http://powershell.com/cs/media/14/default.aspx should work
```powershell
retry of URL failed with error: 
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 111
111:                                 throw "retry of URL failed with error: $($_.Message)"

```

**Comment by daxian-dbw** [comment]
The static analysis failure is unrelated to this PR and a fix is coming in a separate PR.
```none
[-] http://powershell.com/cs/blogs/ebookv2/default.aspx should work 92ms
        RuntimeException: retry of URL failed with error: 
        at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 111
```

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**